documentation/coding_guidelines.md

```diff
@@ -186,7 +186,7 @@ Error handling suggestions follow the [Rust book guidance](https://doc.rust-lang
 
 *Panic*
 
-* `panic!()` - Runtime panic! should only be used when the resulting state cannot be processed going forward.  It should not be used for any recoverable errors.
+* `unrecoverable!()` - Logs an error! causes a panic! Should only be used when the resulting state cannot be processed going forward.
```

Comment:
I'm in support of this change, but to have even more impact I think we should lint panics and point developers to use unrecoverable instead. I can try to do this, would x lint be a good place to do this? @bmwill